packages/perseus/src/widgets/interactive-graphs/mafs-graph.test.tsx

```diff
@@ -63,3 +68,192 @@ describe("MafsGraph", () => {
+
+function map(
+    num: number,
+    startMin: number,
+    startMax: number,
+    targetMin: number,
+    targetMax: number,
+): number {
+    return (
+        ((num - startMin) / (startMax - startMin)) * (targetMax - targetMin) +
+        targetMin
+    );
+}
+
+describe("MafsGraph", () => {
+    let userEvent: UserEvent;
+    beforeEach(() => {
+        userEvent = userEventLib.setup({
+            advanceTimers: jest.advanceTimersByTime,
+        });
+    });
+
+    it("renders", () => {
+        const mockDispatch = jest.fn();
+        const state: InteractiveGraphState = {
+            type: "segment",
+            hasBeenInteractedWith: true,
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+            snapStep: [0.5, 0.5],
+            coords: [
+                [
+                    [0, 0],
+                    [-7, 0.5],
+                ],
+            ],
+        };
+
+        const baseMafsGraphProps = getBaseMafsGraphProps();
+
+        render(
+            <MafsGraph
+                state={state}
+                dispatch={mockDispatch}
+                {...baseMafsGraphProps}
+            />,
+        );
+
+        const line = screen.getByTestId("movable-line__line");
+
+        expect(line).toBeInTheDocument();
+        expect(line.getAttribute("x1")).toBe("0");
+        expect(line.getAttribute("y1")).toBe("0");
+
+        // Map graph coordinates to SVG coordinates
+        const halfXBox = baseMafsGraphProps.box[0] / 2;
+        const halfYBox = baseMafsGraphProps.box[1] / 2;
+        const expectedX2 = map(-7, -10, 10, halfXBox * -1, halfXBox);
+        const expectedY2 = map(0.5, -10, 10, halfYBox * -1, halfYBox) * -1;
+        expect(line.getAttribute("x2")).toBe(`${expectedX2}`);
+        expect(line.getAttribute("y2")).toBe(`${expectedY2}`);
+    });
+
+    /**
+     * regression LEMS-1885
+     * Important parts of this test:
+     * - large snap setting (>1)
+     * - arrowing with the keyboard moves point
+     */
+    it("point snap works with large snap gaps", async () => {
+        const mockDispatch = jest.fn();
+        const state: InteractiveGraphState = {
+            type: "point",
+            hasBeenInteractedWith: true,
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+            snapStep: [2, 2],
+            coords: [[2, 2]],
+        };
+
+        const baseMafsGraphProps = getBaseMafsGraphProps();
+
+        const {rerender} = render(
+            <MafsGraph
+                state={state}
+                dispatch={mockDispatch}
+                {...baseMafsGraphProps}
+            />,
+        );
+
+        const group = screen.getByTestId("movable-point");
+        group.focus();
+        await userEvent.keyboard("[ArrowRight]");
+        const action = movePoint(0, [4, 2]);
+        expect(mockDispatch).toHaveBeenCalledWith(action);
+
+        const updatedState = interactiveGraphReducer(state, action);
```

Comment:
Since the dispatch is mocked, I'm manually recalculating state and rerendering.

Reply:
I think you're right, but I think I'm going to keep this as-is if for no other reason than to show how nuanced we can get.